src/main/java/com/singlestore/jdbc/DatabaseMetaData.java

```
@@ -68,14 +68,42 @@ private static String DataTypeClause(Configuration conf) {
     }
 
     if (!conf.yearIsDateType()) {
-      return " IF(c.COLUMN_TYPE IN ('year(2)', 'year(4)'), 'SMALLINT', "
-          + upperCaseWithoutSize
-          + ")";
+      upperCaseWithoutSize =
+          " IF(c.COLUMN_TYPE IN ('year(2)', 'year(4)'), 'SMALLINT', " + upperCaseWithoutSize + ")";
+    }
+    if (!isExtendedTypesEnabled()) {
+      upperCaseWithoutSize =
+          " IF(c.COLUMN_TYPE LIKE 'vector%', "
+              + (isBinaryVectorOutputEnabled() ? "'VARBINARY'" : "'VARCHAR'")
+              + ", "
+              + upperCaseWithoutSize
+              + ")";
+      upperCaseWithoutSize =
+          " IF(c.COLUMN_TYPE LIKE 'bson%', " + "'LONGBLOB'" + ", " + upperCaseWithoutSize + ")";
     }
-
     return upperCaseWithoutSize;
   }
 
+  /** Is extended types enabled, like VECTOR, BSON, etc... */
+  private boolean isExtendedTypesEnabled() {
+    try {
+      return getSingleStoreVersion().versionGreaterOrEqual(8, 7, 1)
+          && conf.enableExtendedDataTypes();
+    } catch (SQLException e) {
+      return false;
```

Let's log this error rather than silently ignoring it.